### PR TITLE
feat: [CDS-71447]: should not render interruptType AbortAll and UserMarkedFailure at execution header if stageId is not present

### PR DESCRIPTION
**Pull request by Eric761**
#### Video

Before:

https://github.com/harness/harness-core-ui/assets/73639897/2481b4b7-1760-4f6b-adb5-f670353420ec

After:

https://github.com/harness/harness-core-ui/assets/73639897/c8b44781-5efc-4fbe-9d2a-258116bb5f30

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pankajsingh-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx [hunk 1]
```diff
@@ -112,76 +107,44 @@ describe('<ExecutionActions /> tests', () => {
...
-            refetch={jest.fn()}
-          />
-        </TestWrapper>
-      )
-    })
+      let result: RenderResult
 
-    act(() => {
-      const btn = result!.container.querySelector(`[data-icon="${icon}"]`)?.closest('button')
+      act(() => {
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx:125]: remove this `act`

File: src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx [hunk 2]
```diff
@@ -112,76 +107,44 @@ describe('<ExecutionActions /> tests', () => {
...
+              executionStatus={executionStatus}
+              refetch={jest.fn()}
+            />
+          </TestWrapper>
+        )
+      })
 
-      // UserMarkedFailure interrupt is not allowed at pipeline level
-      if (interruptType === 'UserMarkedFailure') {
+      act(() => {
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx:138]: this `act` can be removed

**Review by rameshx — DISMISSED** [review]

File: src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx (hunk 2 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx]: done

File: src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx (hunk 1 above)
Comment (Eric761) [reply src/modules/70-pipeline/components/ExecutionActions/__tests__/ExecutionActions.test.tsx]: done

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by Eric761** [comment]
Unrelated cypress test case is failing

**Pull request merged**